src/lib.rs

```diff
@@ -8,15 +8,30 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-//! Utilities for handling sockets
+// # Source code structure.
+//
+// All types and methods that are available on tier 1 all platforms are defined
+// in the first level of the source, i.e. `src/*.rs` files. Additional API that
+// is platform specific, e.g. `Domain::UNIX`, is defined in `src/sys/*.rs` and
+// only for the platforms that support it.
+
+//! Utilities for creating and using sockets.
+//!
+//! The goal of this crate is to create and use a socket using advanced
+//! configuration options (those that are not available in the types in the
+//! standard library) without using any unsafe code.
+//!
+//! This crate provides as direct as possible access to the system's
+//! functionality for sockets as possible, this means **no** cross-platform
+//! utilities, no extra goodies, no creature comforts. It is up to the user to
```

Comment:
Well... we *do* get plenty of cross-platform utilities as well as platform-specific options :) But I agree with the sentiment.

Reply:
I've changed the wording a bit. Could you take another look?

If you ok with it I'll squash the commits and merge.